Spark: Fix jmh Spark parquet benchmark

zhongyujiang opened the pull request:
This PR fixs jmh Spark parquet benchmark, now the SparkParquetWritersFlatDataBenchmark throws the following exception:
```
# Warmup Iteration   2: <failure>

org.apache.iceberg.exceptions.AlreadyExistsException: File already exists: /Users/zhongyujiang/projects/github/iceberg/spark/v3.2/spark/build/tmp/jmh/parquet-flat-data-benchmark6286636693589332554.parquet
 at org.apache.iceberg.Files$LocalOutputFile.create(Files.java:58)
 at org.apache.iceberg.parquet.ParquetIO$ParquetOutputFile.create(ParquetIO.java:148)
 at org.apache.parquet.hadoop.ParquetFileWriter.<init>(ParquetFileWriter.java:329)
 at org.apache.parquet.hadoop.ParquetFileWriter.<init>(ParquetFileWriter.java:305)
 at org.apache.parquet.hadoop.ParquetFileWriter.<init>(ParquetFileWriter.java:284)
 at org.apache.iceberg.parquet.ParquetWriter.ensureWriterInitialized(ParquetWriter.java:114)
 at org.apache.iceberg.parquet.ParquetWriter.flushRowGroup(ParquetWriter.java:200)
 at org.apache.iceberg.parquet.ParquetWriter.close(ParquetWriter.java:233)
 at org.apache.iceberg.spark.data.parquet.SparkParquetWritersFlatDataBenchmark.writeUsingIcebergWriter(SparkParquetWritersFlatDataBenchmark.java:105)
 at org.apache.iceberg.spark.data.parquet.jmh_generated.SparkParquetWritersFlatDataBenchmark_writeUsingIcebergWriter_jmhTest.writeUsingIcebergWriter_ss_jmhStub(SparkParquetWritersFlatDataBenchmark_writeUsingIcebergWriter_jmhTest.java:416)
 at org.apache.iceberg.spark.data.parquet.jmh_generated.SparkParquetWritersFlatDataBenchmark_writeUsingIcebergWriter_jmhTest.writeUsingIcebergWriter_SingleShotTime(SparkParquetWritersFlatDataBenchmark_writeUsingIcebergWriter_jmhTest.java:371)
```
This PR set the level of `TearDown` to `Level.Invocation` to clean up `dataFile`  for each benchmark method execution.
@aokolnychyi could you take a look? thx.

nastra commented on spark/v3.2/spark/src/jmh/java/org/apache/iceberg/spark/data/parquet/SparkParquetWritersFlatDataBenchmark.java:
```diff
@@ -85,7 +86,7 @@ public void setupBenchmark() throws IOException {
   }
 
-  @TearDown
+  @TearDown(Level.Invocation)
```
not sure but I think both should be Iteration rather than Invocation

zhongyujiang replied:
Iteration and Invocation should have the same effect since BenchmarkMode here is SingleShotTime, but I think is should be Invocation since each Invocation will produce a file.

zhongyujiang commented on .github/workflows/jmh-bechmarks.yml:
```diff
@@ -76,6 +76,9 @@ jobs:
     - uses: actions/checkout@v2
+      with:
+        repository: ${{ github.event.inputs.repo }}
+        ref: ${{ github.event.inputs.ref }}
```
@nastra I run this in my fork, but the [result](https://github.com/zhongyujiang/iceberg/actions/runs/1733298769) shows  it didn't use the ref i chose. And I found job `run-benchmark`  haven't specify repo and ref here.

zhongyujiang replied:
And this [run](https://github.com/zhongyujiang/iceberg/actions/runs/1733299718)  shows benchmarks are working.

nastra replied:
good catch!

rdblue commented:
Running CI. @nastra, do you think this is ready to commit?

nastra reviewed (approved):
LGTM

zhongyujiang commented:
Hi @rdblue, can you help merge this? Or do you have any other comments?

rdblue commented:
@zhongyujiang, @nastra, does this need to be rebased now that $4149 is in? I think that fixed some of the same issues.

zhongyujiang commented:
I have rebased and backported this fix, and changed level from `Level.Invocation` to `Level.Iteration` to be consistent with #4149, both have the same effect here. Please check again. @nastra @rdblue

nastra reviewed (approved):
LGTM but you might want to update your PR title, since it mentions Spark 3.2 only

zhongyujiang commented:
Thanks for reminding, updated.
